lib/model/rwfolder.go

```diff
@@ -784,10 +781,19 @@ func (f *sendReceiveFolder) deleteDir(file protocol.FileInfo, matcher *ignore.Ma
 
 	realName, err := rootedJoinedPath(f.dir, file.Name)
 	if err != nil {
-		f.newError(file.Name, err)
+		f.newError(file.Name, "deleteDir rj", err)
 		return
 	}
 
+	if _, serr := os.Lstat(realName); !os.IsNotExist(serr) {
+		// Verify that the thing we are handling lives inside a directory,
+		// and not a symlink or empty space.
+		if !osutil.IsDir(f.dir, filepath.Dir(file.Name)) {
+			f.newError(file.Name, "deleteDir isDir", errNotDir)
```

Comment:
The delete was successful at this point, the error is pointless.
E.g. ``foo/bar`` was renamed to ``baz/bar`` and a Symlink ``foo -> bar`` was created. The ``Lstat`` check succeeds but the real file ``foo/bar`` doesn't exist anymore.